glusterd2/middleware/auth.go

```diff
@@ -80,7 +82,7 @@ func Auth(next http.Handler) http.Handler {
 
 		// Check if token is Valid
 		if err != nil || !token.Valid {
-			http.Error(w, err.Error(), http.StatusUnauthorized)
+			restutils.SendHTTPError(ctx, w, http.StatusUnauthorized, err.Error())
```

Comment:
just send err instead of err.Error() and in this case the error can be nil. I think the if condition should have && instead of ||, or send a different error if error is nil.

Reply:
That's a good catch! If `err` is nil, `err.Error()` will crash.

Reply:
if ```err is nil``` am not in favor of sending nil response in this case
am not sure when we can hit this issue, but for the safer side, I have separated out the checks.

Reply:
I actually think it maybe impossible to hit based on some similar work I did in heketi. It seems to be a quirk of the jwt library's api that it has this "extra" property but I think that in normal use of the api you can't hit this as you'll always get non-nil error from the parse function (IIRC). It made getting full code coverage in this area in heketi impossible! :-o